external-crates/move/crates/move-compiler/src/hlir/translate.rs

```diff
@@ -1251,48 +1251,134 @@ fn value_block(
     }
 }
 
+// This is an slightly sub-optimal implementation of `value_list`, which we preserve in order to
+// accurately re-produce code that is already on-chain. When we support SHA-based source
+// verification, we should swap this version out with the optimized version below it. The main
+// difference is that this version does conversion and binding, then freezing; the other will
+// inline freezing when possible to avoid some bndings.
+
 fn value_list(
     context: &mut Context,
-    block: &mut Block,
+    result: &mut Block,
```

Comment:
Would it be too much to gate this?

Reply:
I will try gating it, plus the other optimization.